packages/core/src/anchor/ethereum/ethereum-anchor-validator.ts

```diff
@@ -93,10 +110,20 @@ export class EthereumAnchorValidator implements AnchorValidator {
   ): Promise<[TransactionResponse, Block]> {
     try {
       // determine network based on a chain ID
+      console.log(`chainId: ${chainId} : ${txHash} : ${typeof(txHash)}`)
+      console.log("this._getEthProvider(chainId)")
+      console.log(this._getEthProvider(chainId))
       const provider: providers.BaseProvider = this._getEthProvider(chainId)
+      console.log("provider")
+      console.log(provider)
       let transaction = this._transactionCache.get(txHash)
+      console.log("transaction x")
+      console.log(transaction)
 
       if (!transaction) {
```

Comment:
This thing is better moved to a separate function.

Reply:
`if (!transaction) {...` that puts the transaction in the cache should also be there in `_getTransaction`.

The goal is to hide the caching.